tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala

```diff
@@ -22,6 +22,8 @@ class TMailCanSendFrom @Inject()(rrt: RecipientRewriteTable, aliasReverseResolve
           case _: TeamMailboxNotFoundException => SMono.just(false)
           case e => SMono.error(e)
         }
+        .asJava()
+        .share()
```

Comment:
Why do we need `.share` ? What is this operator doing?

Reply:
It is some kind of caching the Mono/turn it to a hot Publisher.
rf: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#share--
https://stackoverflow.com/questions/67534602/webtestclient-returning-illegalstateexception-block-blockfirst-blocklast

Reply:
Do we need to turn into a hot publisher just before blocking???

Reply:
Not really. Actually, this .`share` use is dangerous. This cause `DistributedTeamMailboxesTest` sometimes fail sometimes green. Maybe we need to reactify `userCanSendFrom` at the end.

Reply:
> reactify userCanSendFrom

Adapted a part for TMailCanSendFrom on TMail side and leveraged it in EmailSubmissionSetMethod. I will check where else to leverage in James and push a PR.